### PR TITLE
[RNMobile] Add scrollable screens to the bottom-sheet

### PR DESCRIPTION
**Pull request by dratwas**
GB-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2741

Motivation:
Since we have navigation inside the bottom-sheet, there is a possibility to have a flat list on some of the screens. In that scenario, we need to remove the scroll view from the bottom-sheet dynamically while switching the screen which generates some issues.

In this PR I moved the ScrollView to the navigation-screen and we can set `isScrollable` prop for each screen. I left the ScrollView only for bottom-sheets w/o navigation inside.

I added the `isScrollable` prop to the `BottomSheet.NavigationScreen` which determines if the content of the screen has FlatList/List/ScrollView inside.

## How has this been tested?
- Open Button Settings and check if the scroll works as before
- Open Menu Inserter and check if the scroll works as before
- Open Alignment or Header type bottom-sheet and check if the scroll works as before
- Open link settings bottom-sheet and check if it works as before

This PR shouldn't bring any changes in the way the editor works - it is only a different implementation. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by lukewalczak** [comment]
Hello @dratwas! I was testing the PR on Android and discovered one regression in slider spacing:

master | PR
--- | ---
 | 

I've noticed also wrong behavior in android back button functionality, however I'm able to reproduce it on master as well and so prolly it should be handled on the separate PR - android hardware back button should move user back to the previous sheet instead of to the main
backbutton_bug

**Review by mkevins — APPROVED** [review]
This looks good! I tested the flows in the description via Pixel 3a, and everything is working as expected. Nice work!

I left a minor suggestion for a variable rename, but it is not a blocker.

**Edit**
Oh, just saw Luke's comment (I guess my browser tab was not refreshed). Let's see about resolving the spacing regression before merging.

File: packages/block-editor/src/components/block-settings/container.native.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ function BottomSheetSettings( {
 			onClose={ closeGeneralSidebar }
 			hideHeader
 			contentStyle={ styles.content }
+			withNavigation
```
[the remark this answers is not in the document]
Comment (dratwas) [reply packages/block-editor/src/components/block-settings/container.native.js:30]: You're totally right! Changing :)

**Comment by dratwas** [comment]
Hey, @lukewalczak I fixed the spacing in the slider component. Could you please check it one more time? :)

File: packages/block-editor/src/components/inserter/menu.native.js [hunk 2]
```diff
@@ -154,6 +154,14 @@ export class InserterMenu extends Component {
 								keyExtractor={ ( item ) => item.name }
 								renderItem={ this.renderItem }
 								{ ...listProps }
+								contentContainerStyle={ [
+									...listProps.contentContainerStyle,
+									{
+										paddingBottom:
+											safeAreaBottomInset ||
+											styles.list.paddingBottom,
```
Comment (lukewalczak) [inline packages/block-editor/src/components/inserter/menu.native.js:162]: Not a blocker, just quick idea: can you move `styles.list.paddingBottom` as default value to `safeAreaBottomInset` when destructuring it?

File: packages/components/src/mobile/bottom-sheet/bottom-sheet-navigation/navigation-screen.native.js [hunk 3]
```diff
@@ -69,10 +77,28 @@ const BottomSheetNavigationScreen = ( { children, fullScreen } ) => {
...
+			<ScrollView { ...listProps }>
+				<TouchableHighlight accessible={ false }>
+					<View onLayout={ onLayout }>
+						{ children }
+						{ ! isNested && (
+							<View
+								style={ {
+									height:
+										safeAreaBottomInset ||
+										styles.scrollableContent.paddingBottom,
```
Comment (lukewalczak) [inline packages/components/src/mobile/bottom-sheet/bottom-sheet-navigation/navigation-screen.native.js:93]: The same [idea](https://github.com/WordPress/gutenberg/pull/26204/files#r516549256) we can apply there

File: packages/components/src/mobile/link-picker/index.native.js [hunk 4]
```diff
@@ -81,48 +81,50 @@ export const LinkPicker = ( {
...
+					valueStyle={ styles.omniInput }
+					value={ value }
+					placeholder={ __( 'Search or type URL' ) }
+					autoCapitalize="none"
+					autoCorrect={ false }
+					keyboardType="url"
+					onChangeValue={ setValue }
+					onSubmit={ onSubmit }
+					/* eslint-disable-next-line jsx-a11y/no-autofocus */
+					autoFocus={ true }
```
Comment (lukewalczak) [inline packages/components/src/mobile/link-picker/index.native.js:104]:
```suggestion
					autoFocus
```

**Review by lukewalczak — APPROVED** [review]
Checked again spacing on Android and right now it looks properly 🎉 Left some minor comments - not a blocker.

**Pull request merged**